### ISSUE TITLE
Duplicate / Restore events doubled

### ISSUE THREAD

**Issue by Syxton** [davosmith/moodle-checklist#56]
@davosmith 
Create a checklist with dates shown in the calendar.
Add a single item with a date.
See that the calendar now shows 1 event for that item.
Duplicate the checklist.
See that the calendar now shows 3 events.  1 for the first checklist and 2 for the duplicate checklist.

I have found that commenting out checklist_refresh_events() in lib.php solves this issue, but I'm unsure what else this function does.

**Comment by davosmith** [comment]
Hi Matt,

checklist_refresh_events() creates calendar events for any checklist items that don't already have a link to a calendar item (where the item has a date and calendar events are enabled for the checklist). Unfortunately, the Moodle backup + restore process already creates copies of the calendar events, but my checklist doesn't know anything about them (so if you edited the date on the item, it wouldn't be able to update the calendar properly).

I've added an extra step in the refresh to look for any calendar events associated with a particular checklist that I don't know anything about and delete them.

If you could test the code I've pushed to github and confirm that it fixes the problem, then I would be grateful (I'll push it out to Moodle.org when I get a chance).

**Comment by Syxton** [comment]
@davosmith I believe that fixed the issue.  Only thing I did notice was that if you have an event with a date and they are being shown in the calendar and you turn that setting off in the checklist (to no longer show items in the calendar), the checklist_items still reference an eventid that is deleted instead of being set to 0.  Just a small thing, other than that, it looks like my issue is fixed.  Thanks!

**Comment by davosmith** [comment]
Odd - the code is supposed to go through and delete all the events it knows about, whilst, at the same time, removing the reference from the checklist item.

I'll try and take a look (but not tonight, as it's getting too late to be writing code).

**Comment by Syxton** [comment]
@davosmith Did you find anything.  Am I mistaken?  Thanks for your time on this.

**Comment by davosmith** [comment]
@Syxton I only rarely have spare time to look at my plugins, so this morning is the first free moment I've had to work on this since last week.

You are right that the field is not correctly cleared.

The only problem that causes, as far as I can tell, is an extra DB query at the point when calendar events are turned back on again (as opposed to an extra DB update when they are turned off). However, it is technically incorrect, so I've fixed it.
